app/src/components/blocks/_lloydGeorge/lloydGeorgeSelectDownloadStage/LloydGeorgeSelectDownloadStage.tsx

```diff
@@ -119,9 +117,15 @@ function LloydGeorgeSelectDownloadStage({
             mounted.current = true;
             void onPageLoad();
         }
+        numberOfFilesForDownload.current = selectedDocuments?.length
+            ? selectedDocuments.length
+            : searchResults.length
+            ? searchResults.length
+            : numberOfFilesForDownload.current;
```

Comment:
Putting this calculation in useEffect might give unexpected result in some case, 
for example, when I have 5 files and selected only 3 files, then click the "download-all" button (I tested this on a sandbox), it seems to display "preparing 3 files", which isn't correct (as I downloaded all 5 files).

Maybe we better not use a ref to store this value. Just use a normal variable, and do the calculation out of useEffect, for example:
```
   // replace line 81 `let numberOfFilesForDownload = useRef(numberOfFiles)` with below:
    const numberOfFilesForDownload =
        selectedDocuments?.length || searchResults.length || numberOfFiles;
```

![demo file number issue](https://github.com/nhsconnect/national-document-repository/assets/127404525/e54d51be-c3a2-475a-a502-27363b802e8e)